bee-network/src/service/service.rs

```diff
@@ -0,0 +1,486 @@
+            if relation.is_known() {
+                add_peer(peer_id, multiaddr, alias, relation, senders, peerlist).await?;
+
+                // We automatically connect to such peers. Since we can connect concurrently, we spawn a task here.
+                let _ = senders.internal_commands.send(Command::DialPeer { peer_id });
+            } else {
+                add_peer(peer_id, multiaddr, alias, relation, senders, peerlist).await?;
+            }
+        }
+
+        Command::BanAddress { address } => {
+            let mut peerlist = peerlist.0.write().await;
+            peerlist.ban_address(address.clone())?;
+
+            let _ = senders.events.send(Event::AddressBanned { address });
+        }
+
+        Command::BanPeer { peer_id } => {
+            let mut peerlist = peerlist.0.write().await;
+            peerlist.ban_peer(peer_id)?;
+
+            let _ = senders.events.send(Event::PeerBanned { peer_id });
+        }
+
+        Command::ChangeRelation { peer_id, to } => {
+            let mut peerlist = peerlist.0.write().await;
+            peerlist.update_info(&peer_id, |info| info.relation = to)?;
+        }
+
+        Command::DialAddress { address } => {
+            let _ = senders.internal_commands.send(Command::DialAddress { address });
+        }
+
+        Command::DialPeer { peer_id } => {
+            let _ = senders.internal_commands.send(Command::DialPeer { peer_id });
+        }
+
+        Command::DisconnectPeer { peer_id } => {
+            disconnect_peer(peer_id, senders, peerlist).await?;
+        }
+
+        Command::RemovePeer { peer_id } => {
+            remove_peer(peer_id, senders, peerlist).await?;
+        }
+
+        Command::UnbanAddress { address } => {
+            let mut peerlist = peerlist.0.write().await;
+            peerlist.unban_address(&address)?;
+
+            let _ = senders.events.send(Event::AddressUnbanned { address });
+        }
+
+        Command::UnbanPeer { peer_id } => {
+            let mut peerlist = peerlist.0.write().await;
+            peerlist.unban_peer(&peer_id)?;
+
+            let _ = senders.events.send(Event::PeerUnbanned { peer_id });
+        }
+    }
+
+    Ok(())
+}
+
+async fn process_internal_event(
+    int_event: InternalEvent,
+    senders: &Senders,
+    peerlist: &PeerList,
+) -> Result<(), PeerError> {
+    match int_event {
+        InternalEvent::AddressBound { address } => {
+            let _ = senders.events.send(Event::AddressBound { address });
+        }
+
+        InternalEvent::ProtocolDropped { peer_id } => {
+            let mut peerlist = peerlist.0.write().await;
+
+            // Try to disconnect, but ignore errors in-case the peer was disconnected already.
+            let _ = peerlist.update_state(&peer_id, |state| state.set_disconnected());
+
+            // Try to remove unknown peers.
+            let _ = peerlist.remove_if(&peer_id, |peer_info, _| peer_info.relation.is_unknown());
```

Comment:
Would `retain` be a more consistent name for this API?

Reply:
I changed those methods already to:
`filter`, `filter_count`, and `filter_remove`.